### PR TITLE
fix: #1466 additionalProperties works now in spring generator

### PR DESCRIPTION
**Pull request by jameswynn**
When additional properties is true, use an additionalProperties attribute and @JsonAnySetter and @JsonAnyGetter instead of inherit from Map. This is an updated PR based on @benfonty's work in https://github.com/OpenAPITools/openapi-generator/pull/10610

This fixes https://github.com/OpenAPITools/openapi-generator/issues/1466 only for spring target.

@wing328 @bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by loiccara** [comment]
Hi @benfonty, this issue is blocking my team as well. Is there anything blocking this PR from being merged, apart from the obvious conflicts?

@jameswynn Can you please resolve the conflicts so that this can be released?

**Comment by tlehmannmgm** [comment]
Any chance to get some progress here? The issue is blocking my team as well :-(

File: modules/openapi-generator/src/main/resources/JavaSpring/additionalProperties.mustache
Comment (Zomzog) [inline modules/openapi-generator/src/main/resources/JavaSpring/additionalProperties.mustache:15]: I think the indentation can be reworked

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java
Comment (Zomzog) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java:542]: I don't think it must be set to true by default.

**Comment by hspereboom** [comment]
Just ran into this myself, would be great to have this.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java
Comment (bob-walters) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java:542]: This is consistent with the way that the Java generator uses this flag when it is run with a template library that works with jackson.  I think in the case of the spring generator, all 3 libraries currently use jackson for object serialization.

**Comment by zsait-clearstreet** [comment]
Can this be merged soon and made available in a release? We ran into the same issue.

**Comment by benfonty** [comment]
Hello, I created this pull request a long time ago and did not receive any approval at the time.
If anybody wants to take it over, feel free.

**Comment by borsch** [comment]
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

File: modules/openapi-generator/src/main/resources/JavaSpring/additionalProperties.mustache
Comment (welshm) [inline modules/openapi-generator/src/main/resources/JavaSpring/additionalProperties.mustache:33]: Should the return be wrapped with an `Optional` if `useOptional` is enabled?

**Review by welshm — APPROVED** [review]
I can put up a unit test PR that validates the map is added once this is merged - I think that would also be helpful to prevent regressions.

**Comment by borsch** [comment]
> I can put up a unit test PR that validates the map is added once this is merged - I think that would also be helpful to prevent regressions.

I've added one test

**Comment by welshm** [comment]
> > I can put up a unit test PR that validates the map is added once this is merged - I think that would also be helpful to prevent regressions.
> 
> 
> 
> I've added one test

Nice!

**Pull request merged**
